R/as_big_trait_table.R

```diff
@@ -1,119 +1,133 @@
 #' }
 #' @importFrom rlang .data
 #' 
-wide_table_format <- function(austraits_all){
+as_wide_table <- function(austraits){
 
 
-    
   ################################################################################
+  # TODO: this updated with next zenodo release
   # Load the trait classification doc - classifies the tissue type and type of trait based on the trait_name data field
-  trait_class = read.csv("data_raw/Trait_classifications_v3.csv")
-  trait_class[is.na(trait_class)] = ""
-  ################################################################################
-  #Define the collapse-cols function
-  collapse_cols <- function(data) {
-    data %>% imap_dfr(~ sprintf("%s='%s'",.y,.x)) %>%
-      unite("text", sep="; ") %>% pull(text)
+  # Exclude this for now -- will be added to definitions file in future release
+  # trait_class = read.csv("data-raw/Trait_classifications_v3.csv")
+  # trait_class[is.na(trait_class)] = ""
+  # trait_class <- trait_class %>% as_tibble()
+  # 
+  # we only need two extra columns from the trait class table - collapsing two category and other_tags cols and renaming them for clarity
+  # x2 <- 
+  #   trait_class %>% dplyr::mutate(
+  #   trait_category = str_c(category, "; ", other_tags) %>% gsub("; $", "", .)
+  # ) %>% 
+  #   dplyr::select(trait_name, tissue, trait_category)
+  # 
+  # Function to collapse columns in sites and contexts into single column
+  process_table <- function(data) {
+    
+    # worker function called intop worklfow below
+    # for a df, combine all column names and values
+    collapse_cols <- function(data) {
+      
+      if(ncol(data) ==0) return(NA_character_)
+      
+      data %>% purrr::imap_dfr(~ sprintf("%s='%s'",.y,.x)) %>%
+        tidyr::unite("text", sep="; ") %>% dplyr::pull(text)
+    }
+    
+    data %>% 
+      tidyr::pivot_wider(names_from = property, values_from = value) %>% 
+      tidyr::nest(data=-any_of(c("dataset_id", "site_name", "context_name", "latitude (deg)", "longitude (deg)"))) %>%
+      dplyr::mutate(site = purrr::map_chr(data, collapse_cols)) %>%
+      dplyr::select(-data) 
   }
   
   ################################################################################
   # Define and adapt each table in the list of austraits to prepare for the wide table format 
-  
+
   # the trait table needs little prep. Rename the value columns as value
-  x1 = austraits$traits %>% 
-    rename(c("trait_value" = "value")) 
-  
-  # we only need two extra columns from the trait class table - collapsing two category and other_tags cols and renaming them for clarity
-  x2 = trait_class %>% mutate(trait_category = str_c(category, "; ", other_tags)) %>% 
-    select(trait_name, tissue, trait_category)
-  
-  x2$trait_category = gsub("; $", "", x2$trait_category)
+  austraits$traits <- 
+    austraits$traits %>% 
+    dplyr::rename(c("trait_value" = "value")) 
   
   # The contexts table needs the contexts collapsed to one context name per site
```

Comment:
What does it mean that we need "one context name per site"? The number of contexts allowed should be independent of the site name. There are often multiple contexts per site.